components/event/event-listener/src/main/java/uk/gov/justice/services/event/buffer/EventBufferInterceptor.java

```diff
@@ -37,8 +39,11 @@ public InterceptorContext process(final InterceptorContext interceptorContext, f
     }
 
     private Stream<InterceptorContext> streamFromEventBufferFor(final InterceptorContext interceptorContext) {
-        return eventBufferService.currentOrderedEventsWith(interceptorContext.inputEnvelope())
-                .map(jsonEnvelope -> copyWithInput(interceptorContext, jsonEnvelope));
+        final Stream<JsonEnvelope> envelopeStream = eventBufferService.currentOrderedEventsWith(interceptorContext.inputEnvelope());
+        final List<JsonEnvelope> envelopes = envelopeStream.collect(Collectors.toList());
+        envelopeStream.close();
```

Comment:
This code is run prior to any possible exception being thrown in the Handles methods...so connections will always be closed before continuing with the interceptor chain